engine/src/main/java/org/hibernate/search/engine/mapper/mapping/building/impl/IndexSchemaFilter.java

```diff
@@ -119,28 +123,36 @@ public boolean isPathIncluded(String relativePath) {
 	 */
 	private boolean isPathIncludedInternal(int relativeDepth, String relativePath,
 			boolean markAsEncountered, boolean includedByChild) {
-		boolean includedByThis = depthFilter.isEveryPathIncludedAtDepth( relativeDepth )
-				|| pathFilter.isExplicitlyIncluded( relativePath );
+		boolean includedByThis = ( depthFilter.isEveryPathIncludedAtDepth( relativeDepth )
+				|| pathFilter.isExplicitlyIncluded( relativePath ) )
+				&& !pathFilter.isExplicitlyExcluded( relativePath );
 
 		boolean includedByParent = true;
 		/*
 		 * The parent can filter out paths that are considered as included by a child,
 		 * by reducing the includeDepth in particular,
 		 * but it cannot include paths that are filtered out by a child.
+		 *
+		 * Note: If we only go into recursion when `includedByThis == true` path tracking of parent nodes will not record
+		 * the route.
```

Comment:
Is this still relevant? I'd expect path tracking in parent nodes to not make the difference between a non-existing path and a path excluded in children, so I don't see how this is important.

Reply:
Adding that check actually helps to make the messages cleaner ([in this test for example](https://github.com/hibernate/hibernate-search/pull/3525#discussion_r1221406802))
But it is breaking a few preexisting tests, so I didn't want to make any changes there.
For example:
`org.hibernate.search.test.embedded.path.validation.TestInvalidPaths#testInvalidDeepSimplePathWithLeadingPrefix`:
expects to get `"Encountered field paths: [notJustAb, b.c, b.c.indexed, b.c.notIndexed]"`
but will only get: `Encountered field paths: [notJustAb, b.c, b.c.indexed].`

Or the path tracking test in wants to see a path that "shouldn't" be there - https://github.com/hibernate/hibernate-search/blob/78c77ae0d0347d8250afa028297f8555283f1d5b/engine/src/test/java/org/hibernate/search/engine/mapper/mapping/building/impl/ConfiguredIndexSchemaManagerNestingContextTest.java#L491

Reply:
Feel free to add the check if it fixes tests with no downside.

>  For example:
> `org.hibernate.search.test.embedded.path.validation.TestInvalidPaths#testInvalidDeepSimplePathWithLeadingPrefix`:
> expects to get `"Encountered field paths: [notJustAb, b.c, b.c.indexed, b.c.notIndexed]"`
> but will only get: `Encountered field paths: [notJustAb, b.c, b.c.indexed].` 

The new behavior seems more correct.

> Or the path tracking test in wants to see a path that "shouldn't" be there - 

Same, the new behavior seems more correct. If the path is excluded at level 2, I don't see why it should be visible in the path tracker for level 1.